snapheka/snapheka.go

```diff
@@ -96,9 +103,13 @@ func (p *hekaPublisher) Publish(contentType string, content []byte, config map[s
 
 	u, err := url.Parse(fmt.Sprintf("%s:%d", config["host"].(ctypes.ConfigValueStr).Value, config["port"].(ctypes.ConfigValueInt).Value))
 	handleErr(err)
+	mappingsFile := ""
+	if mFile, ok := config["mappings-file"]; ok {
+		mappingsFile = mFile.(ctypes.ConfigValueStr).Value
+	}
 
 	// Publish metric data to Heka through TCP
-	shc, _ := NewSnapHekaClient(fmt.Sprintf("tcp://%s", u))
+	shc, _ := NewSnapHekaClient(fmt.Sprintf("tcp://%s", u), mappingsFile)
```

Comment:
We could consider creating the client once (assign it to a global var) and use it across publish requests?  (This would require that the routing strategy be changed to sticky.)